### ISSUE TITLE
MIRAGE Segmentation layer mapping to other datasets

### ISSUE THREAD

**Issue by Atif-Anwer** [j-morano/MIRAGE#3]
Hi!

When using Mirage to infer on bscan images in test sets referenced in `segmentation_benchmark.md`; we get the predicted bscan layer masks aswell.  The generated layer masks have the ten intensity values `[0 , 21, 43, 64, 85, 10, 12, 14, 17, 234]` , that they correspond to the pseudo-labels from OCTExplorer. I hope I calculated these correctly?

In order to evaluate the segmentation performance on the ground truth masks, we need to map them to the respective intensity values for each dataset.  I was able to deduce some partial mapping as given below, however I wanted to request if you can provide the correct mapping for the referred datasets please (AROI, Duke DME, Duke iAMD, GOALS, RETOUCH), as I don't think that I have the correct mappings (as Ophthalmology isn't my area of expertise). That would make it much easier to evaluate the results on the datasets, as well as use for other datasets aswell. 

```python
mapping_mirage_to_aroi = {
        0  : 0,
        21 : 23,
        43 : 23,
        64 : 23,
        85 : 23,
        106: 46,
        128: 46,
        149: 46,
        170: 46,
        191: 46,
        213: 69,
        234: 92
}

mapping_mirage_to_dukeDME_iAMD = {
        0  : 0,
        21 : 25,
        43 : 51,
        64 : 76,
        85 : 102,
        106: 102,
        128: 127,
        149: 153,
        170: 178,
        191: 204,
        213: 204,
        234: 204,
    }

mapping_mirage_to_GOALS = {
        0  : 0,
        21 : 80,
        43 : 80,
        64 : 80,
        85 : 160,
        106: 160,
        128: 0,
        149: 0,
        170: 255,
        191: 255,
        213: 255,
        234: 0,
    }
``` 
**EDIT (23/10):** 
I was able estimate the labels for the Datasets from their papers referenced in the [Supplementary Material](https://static-content.springer.com/esm/art%3A10.1038%2Fs41746-025-01852-3/MediaObjects/41746_2025_1852_MOESM1_ESM.pdf). However, as you can probably see, they do require some correction. So your input in this regard will be highly appreciated! 
Thanks!

**Comment by j-morano** [comment]
Hello! As I mentioned in #2, though MIRAGE was pretrained using pseudo-labels of retinal layers, it is not supposed to be used directly for this task, as it was only employed for improving representation learning. For this reason, for evaluating on the different segmentation datasets, the decoders used for pretraining were discarded, and new CNN-based decoders were tuned for each task. Thus, there is no mapping from the pretraining layer labels to the labels in the evaluation datasets. For the evaluation, the only mapping that we used was for the cross-dataset evaluation, where the encoder(frozen)+decoder(tuned) combination trained on AROI was tested on DukeiAMD.

**Comment by Atif-Anwer** [comment]
Thankyou very much for the clarification! I will look into implementing a decoder adapter head and training it for evaluation.
